src/logger/sentry.ts

```diff
@@ -0,0 +1,40 @@
+import * as Sentry from '@sentry/react-native';
+import { SENTRY_ENDPOINT, SENTRY_ENVIRONMENT } from 'react-native-dotenv';
+import VersionNumber from 'react-native-version-number';
+
+import { IS_PROD } from '@/env';
+
+/**
+ * We need to disable React Navigation instrumentation for E2E tests because
+ * detox doesn't like setTimeout calls that are used inside When enabled detox
+ * hangs and timeouts on all test cases
+ */
+export const sentryRoutingInstrumentation = IS_PROD
+  ? new Sentry.ReactNavigationInstrumentation()
+  : undefined;
+
+export const defaultOptions = {
+  dsn: SENTRY_ENDPOINT,
+  enableAutoSessionTracking: true,
+  environment: SENTRY_ENVIRONMENT,
+  integrations: [
+    new Sentry.ReactNativeTracing({
+      routingInstrumentation: sentryRoutingInstrumentation,
+      tracingOrigins: ['localhost', /^\//],
+    }),
+  ],
+  tracesSampleRate: 0.2,
+};
+
+export async function initSentry() {
+  if (!IS_PROD) return;
```

Comment:
Double checked that `Sentry` doesn't care if we call other methods on it before we've called `init()`. So might as well not even init in dev/test.

We're also protected by the fact that the Sentry "transport" isn't attached to the logger in dev/test, but this felt like a good thing to have too.

Reply:
what env are our test flight builds in?

Reply:
For posterity: TF is prod. That's fine for now since it's how it's been working, but we can improve this in the future.